hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -773,37 +773,35 @@ public boolean merge(MapMergeTypes mergingEntry, SplitBrainMergePolicy<Data, Map
             storage.put(key, record);
             eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
                     key, null, record.getValue());
-        } else {
-            oldValue = record.getValue();
-            MapMergeTypes existingEntry = createMergingEntry(serializationService, record);
-            newValue = mergePolicy.merge(mergingEntry, existingEntry);
-            // existing entry will be removed
-            if (newValue == null) {
-                removeIndex(record);
-                mapDataStore.remove(key, now);
-                onStore(record);
-                eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(),
-                        partitionId, key, oldValue, null);
-                storage.removeRecord(record);
-                return true;
-            }
-            if (newValue == mergingEntry.getValue()) {
-                mergeRecordExpiration(record, mergingEntry);
-            }
-            // same with the existing entry so no need to map-store etc operations.
-            if (recordComparator.isEqual(newValue, oldValue)) {
-                return true;
-            }
-            newValue = mapDataStore.add(key, newValue, now);
+            saveIndex(record, null);
+            return true;
+        } else if (newValue == null) {
+            mapDataStore.remove(key, now);
             onStore(record);
             eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
```

Comment:
It's still `writeUpdateEvent`, I think it should be `eventJournal.writeRemoveEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId, key, existingEntry.getValue());`